src/Pages/PlaylistVideos/PlaylistVideos.jsx

```diff
@@ -1,14 +1,55 @@
-import React,{useEffect,useState} from 'react'
-import VideoCard from '../../Components/VideoCard/VideoCard';
+import React, { useEffect, useState } from "react";
+import VideoCard from "../../Components/VideoCard/VideoCard";
+import { useUserData } from "../../Context/UserData-Context";
+import { useNavigate } from "react-router-dom";
 import axios from "axios";
+import { Navigate } from "react-router-dom";
 
 const PlaylistVideos = () => {
-  
+  const [playlistVideos, setPlaylistVideos] = useState([]);
+  const [playlistData, setPlaylistData] = useState({});
+  const {
+    playlistRender,
+    setPlaylistRender,
+  } = useUserData();
+  const playlistUrl = `${window.location.pathname}`.slice(10);
+  useEffect(() => {
+    (async () => {
+      const response = await axios.get(`/api/user/playlists/${playlistUrl}`, {
+        headers: {
+          authorization: JSON.parse(localStorage.getItem("user")).encodedToken,
```

Comment:
Suggestion : can get data from local storage in auth context and set it in a state so that you dont have to extract local storage data in all components.